### ISSUE TITLE
Angular cli error on components in spec file upon ng serve

### ISSUE THREAD

**Issue by DanPride** [angular/angular-cli#1332]
Mac OSX Yosemite
angular-cli: 1.0.0-beta.9
node: 6.3.0
os: darwin x64

Using Angular-cli to create components is generating a consistent error on the .spec files on line 17

//let component = new TodoListComponent() // expect(component).toBeTruthy();

The Broccoli Plugin: [BroccoliTypeScriptCompiler] failed with:

Error: Typescript found the following errors:
  /Users/dp/thechariot/eggAng1/tmp/broccoli_type_script_compiler-input_base_path-UVT171nw.tmp/0/src/app/todo-list/todo-list.component.spec.ts (20, 1): '}' expected.
    at BroccoliTypeScriptCompiler._doIncrementalBuild (/Users/dp/thechariot/eggAng1/node_modules/angular-cli/lib/broccoli/broccoli-typescript.js:120:19)
    at BroccoliTypeScriptCompiler.build (/Users/dp/thechariot/eggAng1/node_modules/angular-cli/lib/broccoli/broccoli-typescript.js:43:10)
    at /Users/dp/thechariot/eggAng1/node_modules/angular-cli/node_modules/broccoli-caching-writer/index.js:152:21
    at lib$rsvp$$internal$$tryCatch (/Users/dp/thechariot/eggAng1/node_modules/angular-cli/node_modules/rsvp/dist/rsvp.js:1036:16)
    at lib$rsvp$$internal$$invokeCallback (/Users/dp/thechariot/eggAng1/node_modules/angular-cli/node_modules/rsvp/dist/rsvp.js:1048:17)
    at lib$rsvp$$internal$$publish (/Users/dp/thechariot/eggAng1/node_modules/angular-cli/node_modules/rsvp/dist/rsvp.js:1019:11)
    at lib$rsvp$asap$$flush (/Users/dp/thechariot/eggAng1/node_modules/angular-cli/node_modules/rsvp/dist/rsvp.js:1198:9)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)

The broccoli plugin was instantiated at: 
    at BroccoliTypeScriptCompiler.Plugin (/Users/dp/thechariot/eggAng1/node_modules/angular-cli/node_modules/broccoli-plugin/index.js:10:31)
    at BroccoliTypeScriptCompiler.CachingWriter [as constructor] (/Users/dp/thechariot/eggAng1/node_modules/angular-cli/node_modules/broccoli-caching-writer/index.js:21:10)
    at BroccoliTypeScriptCompiler (/Users/dp/thechariot/eggAng1/node_modules/angular-cli/lib/broccoli/broccoli-typescript.js:27:5)
    at Angular2App._getTsTree (/Users/dp/thechariot/eggAng1/node_modules/angular-cli/lib/broccoli/angular2-app.js:329:18)
    at Angular2App._buildTree (/Users/dp/thechariot/eggAng1/node_modules/angular-cli/lib/broccoli/angular2-app.js:124:23)
    at new Angular2App (/Users/dp/thechariot/eggAng1/node_modules/angular-cli/lib/broccoli/angular2-app.js:53:23)
    at module.exports (/Users/dp/thechariot/eggAng1/angular-cli-build.js:10:10)
    at Class.module.exports.Task.extend.setupBroccoliBuilder (/Users/dp/thechariot/eggAng1/node_modules/angular-cli/node_modules/ember-cli/lib/models/builder.js:55:19)
    at Class.module.exports.Task.extend.init (/Users/dp/thechariot/eggAng1/node_modules/angular-cli/node_modules/ember-cli/lib/models/builder.js:89:10)
    at new Class (/Users/dp/thechariot/eggAng1/node_modules/angular-cli/node_modules/core-object/core-object.js:18:12)
    at Class.module.exports.Task.extend.run (/Users/dp/thechariot/eggAng1/node_modules/angular-cli/node_modules/ember-cli/lib/tasks/serve.js:15:19)
    at /Users/dp/thechariot/eggAng1/node_modules/angular-cli/node_modules/ember-cli/lib/commands/serve.js:64:24
    at lib$rsvp$$internal$$tryCatch (/Users/dp/thechariot/eggAng1/node_modules/angular-cli/node_modules/rsvp/dist/rsvp.js:1036:16)
    at lib$rsvp$$internal$$invokeCallback (/Users/dp/thechariot/eggAng1/node_modules/angular-cli/node_modules/rsvp/dist/rsvp.js:1048:17)
    at /Users/dp/thechariot/eggAng1/node_modules/angular-cli/node_modules/rsvp/dist/rsvp.js:331:11
    at lib$rsvp$asap$$flush (/Users/dp/thechariot/eggAng1/node_modules/angular-cli/node_modules/rsvp/dist/rsvp.js:1198:9)

**Comment by filipesilva** [comment]
Seems like you have a syntax error `/src/app/todo-list/todo-list.component.spec.ts (20, 1): '}' expected.`.

#1455 should make these errors much more readable.

**Comment by DanPride** [comment]
I agree but its not my error, its the code generated by cli
 Dan Pride
1-303-800-0900 - Skype No Texting1-360-850-9322 Mobile
http://danielpride.com
www.linkedin.com/in/danielpride/http://archaeolibrary.com/https://headlessuber.com

**Comment by filipesilva** [comment]
Can you show me `todo-list.component.spec.ts` please?

**Comment by MalliswariKulala** [comment]
Even I am facing issue while running ng serve. I have upgraded my project to latest angular and angular-cli to @angular/cli.

When I run ng serve command, I get following error


Time: 18152ms                                                                                                                                                                                
chunk    {0} 0.chunk.js, 0.chunk.js.map 13.9 kB {1} [rendered]                                                                                                                               
chunk    {1} main.bundle.js, main.bundle.js.map (main) 159 kB {3} [initial] [rendered]                                                                                                       
chunk    {2} styles.bundle.js, styles.bundle.js.map (styles) 140 kB {4} [initial] [rendered]                                                                                                 
chunk    {3} vendor.bundle.js, vendor.bundle.js.map (vendor) 5 MB [initial] [rendered]                                                                                                       
chunk    {4} inline.bundle.js, inline.bundle.js.map (inline) 0 bytes [entry] [rendered]                                                                                                      
                                                                                                                                                                                             
ERROR in d:/Project/Cosym/Cosym1.2/mockup/client/src/app/system/connections-viewer/connections-viewer.component.spec.ts (69,9): Type '{ srcPropertyName: string; srcPortName: string; targetP
ropertyName: string; targetPortName: strin...' is not assignable to type 'Connection[]'.                                                                                                     
  Type '{ srcPropertyName: string; srcPortName: string; targetPropertyName: string; targetPortName: strin...' is not assignable to type 'Connection'.                                        
    Property 'uri' is missing in type '{ srcPropertyName: string; srcPortName: string; targetPropertyName: string; targetPortName: strin...'.)                                               
d:/Project/Cosym/Cosym1.2/mockup/client/src/app/system/connections-viewer/connections-viewer.component.spec.ts (83,9): Type '{ srcPropertyName: string; srcPortName: string; targetPropertyNa
me: string; targetPortName: strin...' is not assignable to type 'Connection[]'.                                                                                                              
  Type '{ srcPropertyName: string; srcPortName: string; targetPropertyName: string; targetPortName: strin...' is not assignable to type 'Connection'.                                        
    Property 'uri' is missing in type '{ srcPropertyName: string; srcPortName: string; targetPropertyName: string; targetPortName: strin...'.)                                               
d:/Project/Cosym/Cosym1.2/mockup/client/src/app/system/connections-viewer/connections-viewer.component.spec.ts (94,9): Type '{ srcPropertyName: string; srcPortName: string; targetPropertyNa
me: string; targetPortName: strin...' is not assignable to type 'Connection[]'.                                                                                                              
  Type '{ srcPropertyName: string; srcPortName: string; targetPropertyName: string; targetPortName: strin...' is not assignable to type 'Connection'.                                        
    Property 'uri' is missing in type '{ srcPropertyName: string; srcPortName: string; targetPropertyName: string; targetPortName: strin...'.)                                               
                                                                                                                                                                                             
ERROR in d:/Project/Cosym/Cosym1.2/mockup/client/src/app/system/system-dashboard/system-dashboard.component.spec.ts (93,15): Property 'systemId' does not exist on type 'SystemDashboardCompo
nent'.)                                                                                                                                                                                      
d:/Project/Cosym/Cosym1.2/mockup/client/src/app/system/system-dashboard/system-dashboard.component.spec.ts (105,15): Property 'systemId' does not exist on type 'SystemDashboardComponent'.) 
d:/Project/Cosym/Cosym1.2/mockup/client/src/app/system/system-dashboard/system-dashboard.component.spec.ts (107,15): Property 'startSystem' does not exist on type 'SystemDashboardComponent'
.)                                                                                                                                                                                           
d:/Project/Cosym/Cosym1.2/mockup/client/src/app/system/system-dashboard/system-dashboard.component.spec.ts (114,15): Property 'systemId' does not exist on type 'SystemDashboardComponent'.) 
d:/Project/Cosym/Cosym1.2/mockup/client/src/app/system/system-dashboard/system-dashboard.component.spec.ts (116,15): Property 'startSystem' does not exist on type 'SystemDashboardComponent'
.)                                                                                                                                                                                           
webpack: Failed to compile.  


There is no error actually in my spec files. I have commented code in specs and compiled it worked. Later I have uncommented the code, still it compiled successfully.

I did fresh ng serve again, facing same issue..

Could someone help me with this. Thanks in advance :)
